src/satosa/micro_services/custom_routing.py

```diff
@@ -2,14 +2,117 @@
 from base64 import urlsafe_b64encode
 
 from satosa.context import Context
+from satosa.internal import InternalData
+
 from .base import RequestMicroService
 from ..exception import SATOSAConfigurationError
 from ..exception import SATOSAError
+from ..exception import SATOSAStateError
 
 
 logger = logging.getLogger(__name__)
 
 
+class CustomRoutingError(SATOSAError):
+    """
+    SATOSA exception raised by CustomRouting rules
+    """
+    pass
+
+
+class DecideBackendByTarget(RequestMicroService):
```

Comment:
This micro-service selects
> which backend should be used based on who is the SAML IDP

This should be reflected on the name; I think we should call this `DecideBackendByTargetIdP`

Reply:
fixed here: https://github.com/peppelinux/SATOSA/commit/9ff7422633306646c0bb6170d5fc343026a5d9b8